espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/drivers/ComposeDriver.kt

```diff
@@ -0,0 +1,56 @@
+package io.appium.espressoserver.lib.drivers
+
+import androidx.compose.ui.semantics.SemanticsProperties
+import androidx.compose.ui.test.performClick
+import io.appium.espressoserver.EspressoServerRunnerTest
+import io.appium.espressoserver.lib.handlers.exceptions.InvalidElementStateException
+import io.appium.espressoserver.lib.handlers.exceptions.NoSuchElementException
+import io.appium.espressoserver.lib.helpers.ComposeRequestBuilder
+import io.appium.espressoserver.lib.helpers.ComposeViewCache
+import io.appium.espressoserver.lib.model.AppiumParams
+import io.appium.espressoserver.lib.model.BaseElement
+import io.appium.espressoserver.lib.model.ComposeElement
+import io.appium.espressoserver.lib.model.Locator
+
+class ComposeDriver : AppDriver {
+    override fun findElement(params: Locator): BaseElement {
+        try {
+            EspressoServerRunnerTest.composeTestRule.onNode(ComposeRequestBuilder.locatorToMatcher(params))
```

Comment:
would it make sense if we keep https://developer.android.com/reference/kotlin/androidx/compose/ui/test/SemanticsNodeInteraction in the cache instead and throw StaleElementException instead if the node does not exist?

Reply:
Yes, makes sense. I have changed the whole caching and finding logic around it now